### PR TITLE
[testharness.js][docs] Support assert_throws without name property

### LINKED ISSUE #11556: assert_throws doesn't allow arbitrary JS objects
See https://github.com/web-platform-tests/wpt/pull/11496#discussion_r195466654

cc/ @saschanaz

### PR DESCRIPTION
**Pull request by zcorpan**
Fixes #11556.

### CODE REVIEW

File: resources/test/tests/functional/api-tests-1.html [hunk 1]
```diff
@@ -173,6 +173,19 @@ <h1>Sample HTML5 API Tests</h1>
         assert_throws("TEST_ERR", function() {throw e});
     }, "Test assert_throws with non-DOM-exception expected to Fail");
 
+    function FooError() {
+      return this;
+    }
+    test(function()
+    {
+        assert_throws(new FooError(), function() {throw new FooError()});
```
Comment (saschanaz) [inline resources/test/tests/functional/api-tests-1.html]: Can we allow `assert_throws(FooError, ...);`? Functions also have `name` property so I think it's doable without too much changes.

File: resources/testharness.js [hunk 2]
```diff
@@ -1313,12 +1313,21 @@ policies and contribution forms [3].
...
             }
             if (typeof code === "object") {
-                assert("name" in e && e.name == code.name,
-                       "assert_throws", description,
-                       "${func} threw ${actual} (${actual_name}) expected ${expected} (${expected_name})",
-                                    {func:func, actual:e, actual_name:e.name,
-                                     expected:code,
-                                     expected_name:code.name});
+                if (!("name" in code)) {
+                    assert(e.constructor.name == code.constructor.name,
```
Comment (saschanaz) [inline resources/testharness.js]: Can we do `e.constructor.name === code.name`, to allow all of the following cases:

* `assert_throws(new TypeError(), ...)`
* `assert_throws(TypeError, ...)`
* `assert_throws(FooError, ...)`
* `assert_throws({ name: "FooError" }, ...)` (If the error constructor is not exposed as in webidl2.js case)

**Comment by Ms2ger** [comment]
See #11726 for related discussion.

**Comment by zcorpan** [comment]
Closing this in light of https://github.com/web-platform-tests/rfcs/pull/29

**Pull request closed without merging**